### PR TITLE
Reduces overall memory allocation in DynatraceExporterV2 while exporting

### PR DESCRIPTION
**Pull request by belks**
In applications with a lot of metrics the Dynatrace exporter collects all lines to be exported in one list before sending them in batches. This allocates quite a lot of unnecessary memory.
This PR changes this behavior so that batches are sent immediately and already sent lines are not kept in memory.

### CODE REVIEW

File: implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java [hunk 1]
```diff
@@ -139,9 +136,22 @@ public void export(List<Meter> meters) {
         // Lines that are too long to be ingested into Dynatrace, as well as lines that
         // contain NaN or Inf values are not returned from "toMetricLines", and are
         // therefore dropped.
```
Comment (shakuzen) [inline implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java:138]: This comment should probably also be moved to where `toMetricLines` is being called.

**Review by shakuzen — APPROVED** [review]
Thanks for the pull request. @pirgeo let us know if you have any concerns with this change.

**Review by pirgeo — APPROVED** [review]
No concerns from my side!

File: implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java (hunk 1 above)
Comment (belks) [reply implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java]: Done. Moved the comment closer to ` toMetricLines`

**Pull request merged**